site/pages/api/create-social-media-post.test.ts

```diff
@@ -0,0 +1,290 @@
+        const { req, res } = getMockRequestAndResponse({
+            body: previousCreateSocialMediaPostInformation,
+            mockWriteHeadFn: writeHeadMock,
+        });
+        formParseSpy.mockResolvedValue({
+            fields: previousCreateSocialMediaPostInformation,
+
+            files: [
+                {
+                    size: 90000000,
+                    filepath: "/testPath",
+                    newFilename: "testFile",
+                    mimetype: "image/jpg",
+                    mtime: null,
+                    originalFilename: "",
+                } as formidable.File,
+            ],
+        });
+        await createSocialMediaPost(req, res);
+
+        expect(s3Spy).not.toHaveBeenCalledTimes(1);
+        expect(upsertSocialMediaPostSpy).not.toHaveBeenCalledTimes(1);
+
+        const errors: ErrorInfo[] = [{ errorMessage: "Max image size is 5MB.", id: "image" }];
+
+        expect(setCookieOnResponseObject).toHaveBeenCalledTimes(1);
+
+        expect(setCookieOnResponseObject).toHaveBeenCalledWith(
+            COOKIES_SOCIAL_MEDIA_ERRORS,
+            JSON.stringify({ inputs: previousCreateSocialMediaPostInformation, errors }),
+            res,
+        );
+
+        expect(writeHeadMock).toBeCalledWith(302, {
+            Location: `${CREATE_SOCIAL_MEDIA_POST_PAGE_PATH}/${defaultDisruptionId}/0`,
+        });
+    });
+
+    it("should redirect to /create-social-media-post when the image is the wrong type", async () => {
+        const { req, res } = getMockRequestAndResponse({
+            body: previousCreateSocialMediaPostInformation,
+            mockWriteHeadFn: writeHeadMock,
+        });
+        formParseSpy.mockResolvedValue({
+            fields: previousCreateSocialMediaPostInformation,
+
+            files: [
+                {
+                    size: 5000,
+                    filepath: "/testPath",
+                    newFilename: "testFile",
+                    mimetype: "image/webapp",
+                    mtime: null,
+                    originalFilename: "",
+                } as formidable.File,
+            ],
+        });
+        await createSocialMediaPost(req, res);
+
+        expect(s3Spy).not.toHaveBeenCalledTimes(1);
+        expect(upsertSocialMediaPostSpy).not.toHaveBeenCalledTimes(1);
+
+        const errors: ErrorInfo[] = [{ errorMessage: "Only .jpg, .jpeg and .png formats are supported.", id: "image" }];
+
+        expect(setCookieOnResponseObject).toHaveBeenCalledTimes(1);
+
+        expect(setCookieOnResponseObject).toHaveBeenCalledWith(
+            COOKIES_SOCIAL_MEDIA_ERRORS,
+            JSON.stringify({ inputs: previousCreateSocialMediaPostInformation, errors }),
+            res,
+        );
+
+        expect(writeHeadMock).toBeCalledWith(302, {
+            Location: `${CREATE_SOCIAL_MEDIA_POST_PAGE_PATH}/${defaultDisruptionId}/0`,
+        });
+    });
+
+    it("should redirect to /review-disruption when all required inputs are passed", async () => {
```

Comment:
There are 2 tests with this description: "should redirect to /review-disruption when all required inputs are passed"

Reply:
done